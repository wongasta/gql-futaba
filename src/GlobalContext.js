//@flow
import * as React from "react";
import type {UserObjType} from './App';

export const GlobalContext: React.Context<UserObjType> = React.createContext({
  user: null,
  user_id: null
});