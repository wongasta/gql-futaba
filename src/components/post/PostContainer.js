import React from "react";

function PostContainer(){

}

export default PostContainer;