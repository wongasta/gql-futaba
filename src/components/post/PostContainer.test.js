import React from 'react';
import { mount } from "enzyme";
import { MemoryRouter } from 'react-router'
import graphql from 'babel-plugin-relay/macro';
import toJson from "enzyme-to-json";
import {QueryRenderer} from 'react-relay';
import {GlobalContext} from "../../GlobalContext";
import {
  createMockEnvironment,
  MockPayloadGenerator,
} from 'relay-test-utils';
import PostContainer from "./PostContainer";

describe("PostContainer", ()=>{

  afterAll(()=>{
    jest.clearAllMocks();
  })

  test("Post::initial render", ()=>{
    const mockEnvironment = createMockEnvironment();
    const PostContainerWrapper = mount(<MemoryRouter>
      <GlobalContext.Provider value={{user: "abc", user_id: "abc"}}>
        <QueryRenderer
          environment={mockEnvironment}
          query={graphql`
            query PostContainerTestQuery($id: ID!, $comment_count: Int) @relay_test_operation{
              post(id: $id){
                ...PostContainer_post @arguments(comment_count: $comment_count)
              }
            }
`         }
          variables={{
            id: 'globalid_123',
            comment_count: 2147483647
          }}
          render={({error, props})=>{
            if(!props) return null;
            return <PostContainer post={props.post} />
          }}
        />
      </GlobalContext.Provider>
    </MemoryRouter>);
    mockEnvironment.mock.resolveMostRecentOperation(operation=>
      MockPayloadGenerator.generate(operation, {
        Post(context,generateId){
          return {
            "id": generateId().toString(),
            "user_id": generateId().toString(),
            "title": "Test Title",
            "post_content": "Test Content",
            "image_url": "",
            "created_ts": 1604807877,
            "comments": {
              "edges": [
                {
                  "cursor": generateId().toString(),
                  "node": {
                    "id": generateId().toString(),
                    "user_id": generateId().toString(),
                    "comment_content": "Comment 1",
                    "image_url": "",
                    "created_ts": 1604807884
                  }
                },{
                  "cursor": generateId().toString(),
                  "node": {
                    "id": generateId().toString(),
                    "user_id": generateId().toString(),
                    "comment_content": "Comment 2",
                    "image_url": "",
                    "created_ts": 1604807885
                  }
                }
              ]
            }
          }
        }
      })
    );
    PostContainerWrapper.update();
    const children = PostContainerWrapper.find(PostContainer).find('section').children();
    expect(children.length).toBe(4);
    expect(toJson(PostContainerWrapper.find(PostContainer).find('section'))).toMatchSnapshot();
  });

});