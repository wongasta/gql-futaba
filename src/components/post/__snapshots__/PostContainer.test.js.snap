// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostContainer Post::initial render 1`] = `
<section
  className="container"
>
  <p
    className="preview_header"
  >
    <span
      className="title_display"
    >
      Test Title
       
    </span>
    <span
      className="user_display"
    >
      2
       
    </span>
    <span
      className="ts_display"
    >
      Saturday, November 7th 2020, 9:57:57 pm
    </span>
  </p>
  <div
    className="content_container"
  >
    Test Content
  </div>
  <div
    className="comment_container"
    key="3"
  >
    <p
      className="preview_header"
    >
      <span
        className="user_display"
      >
        5
         
      </span>
      <span
        className="ts_display"
      >
        Saturday, November 7th 2020, 9:58:04 pm
      </span>
    </p>
    <div
      className="content_container"
    >
      Comment 1
    </div>
  </div>
  <div
    className="comment_container"
    key="6"
  >
    <p
      className="preview_header"
    >
      <span
        className="user_display"
      >
        8
         
      </span>
      <span
        className="ts_display"
      >
        Saturday, November 7th 2020, 9:58:05 pm
      </span>
    </p>
    <div
      className="content_container"
    >
      Comment 2
    </div>
  </div>
</section>
`;
